.travis.yml: Update supported Ruby versions

perlun opened the pull request:
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Ensure all commits include DCO sign-off.
- [x] Ensure that your contributions pass unit testing.
- [x] Ensure that your contributions contain documentation if applicable.

### Description
- Drop 1.9, 2.0 and 2.1 which are EOL
- Update 2.3, 2.4 and 2.5 specifiers (only specify minor version, not patch level)
- Update JRuby to use latest and greatest

Because this change is breaking in nature, **this will require a major version bump** to comply with SemVer guidelines.

perlun commented:
@rvodden @tarcinil Please share your thoughts before I'll go ahead and merge this.

tarcinil commented on .travis.yml:
```diff
@@ -1,15 +1,12 @@
+  - 2.3
+  - 2.4
+  - 2.5
+  - jruby-9.2
```
I do not see jruby-9.2 on the list of rubies for Travis and the [build](https://travis-ci.org/ruby-git/ruby-git/jobs/396541130) shows that can't find it. 

Travis Rubies: http://rubies.travis-ci.org/

perlun replied:
Good point, I filed https://github.com/rvm/rvm/pull/4414 about it.

The Travis list usually don't include all the RVM versions, but either way, `jruby-9.2` is not a valid RVM version string in this case.

perlun commented:
@tarcinil 

> The only issue I saw was that jruby-9.2 is not a supported version of JRuby for Travis.

Thanks for the feedback, I fixed the PR and squashed it (to retain the DCO comment). Feel free to re-check and I'll merge once you are fine with this.

tarcinil commented:
![bubbles-good](https://user-images.githubusercontent.com/1433666/41973284-24b985d6-79da-11e8-8588-7434757e332b.jpg)
